planning/mission_planner/src/mission_planner/mission_planner.cpp

```diff
@@ -174,557 +134,262 @@ void MissionPlanner::on_map(const HADMapBin::ConstSharedPtr msg)
-  }
-  if (!normal_route_) {
-    throw component_interface_utils::ServiceException(
-      ResponseCode::ERROR_PLANNER_UNREADY, "Normal route is not set.");
-  }
-  if (mrm_route_) {
-    throw component_interface_utils::ServiceException(
-      ResponseCode::ERROR_INVALID_STATE, "Cannot reroute in the emergency state.");
-  }
-  if (reroute_availability_ && !reroute_availability_->availability) {
-    throw component_interface_utils::ServiceException(
-      ResponseCode::ERROR_INVALID_STATE, "Cannot reroute as the planner is not in lane following.");
+  // Restore planner state that changes with create_route function.
+  if (current_route_) {
+    planner_->updateRoute(*current_route_);
   }
+}
 
-  // set to changing state
-  change_state(RouteState::Message::CHANGING);
+LaneletRoute MissionPlanner::create_route(const SetLaneletRoute::Request & req)
+{
+  const auto & header = req.header;
+  const auto & segments = req.segments;
+  const auto & goal_pose = req.goal_pose;
+  const auto & uuid = req.uuid;
+  const auto & allow_goal_modification = req.allow_modification;
 
-  // Convert request to a new route.
-  const auto new_route = create_route(req);
+  return create_route(header, segments, goal_pose, uuid, allow_goal_modification);
+}
 
-  // Check planned routes
-  if (new_route.segments.empty()) {
-    change_route(*normal_route_);
-    change_state(RouteState::Message::SET);
-    res->status.success = false;
-    throw component_interface_utils::ServiceException(
-      ResponseCode::ERROR_PLANNER_FAILED, "The planned route is empty.");
-  }
+LaneletRoute MissionPlanner::create_route(const SetWaypointRoute::Request & req)
+{
+  const auto & header = req.header;
+  const auto & waypoints = req.waypoints;
+  const auto & goal_pose = req.goal_pose;
+  const auto & uuid = req.uuid;
+  const auto & allow_goal_modification = req.allow_modification;
 
-  // check route safety
-  if (check_reroute_safety(*normal_route_, new_route)) {
-    // success to reroute
-    change_route(new_route);
-    res->status.success = true;
-    change_state(RouteState::Message::SET);
-  } else {
-    // failed to reroute
-    change_route(*normal_route_);
-    res->status.success = false;
-    change_state(RouteState::Message::SET);
-    throw component_interface_utils::ServiceException(
-      ResponseCode::ERROR_REROUTE_FAILED, "New route is not safe. Reroute failed.");
-  }
+  return create_route(header, waypoints, goal_pose, uuid, allow_goal_modification);
 }
 
-// NOTE: The route interface should be mutually exclusive by callback group.
-void MissionPlanner::on_change_route_points(
-  const SetRoutePoints::Service::Request::SharedPtr req,
-  const SetRoutePoints::Service::Response::SharedPtr res)
+LaneletRoute MissionPlanner::create_route(const PoseWithUuidStamped & msg)
 {
-  using ResponseCode = autoware_adapi_v1_msgs::srv::SetRoutePoints::Response;
+  const auto & header = msg.header;
+  const auto & waypoints = std::vector<Pose>();
```

Comment:
no need to use & for temporary object?

Reply:
Fixed with https://github.com/autowarefoundation/autoware.universe/pull/6363/commits/4b5cf34f1b170b0ea64af00cd2559d5e27634ffe.